core/src/main/java/org/elasticsearch/indices/IndicesQueryCache.java

```diff
@@ -103,7 +103,7 @@ public QueryCacheStats getStats(ShardId shard) {
         }
         final double weight = totalSize == 0
                 ? 1d / stats.size()
-                : shardStats.getCacheSize() / totalSize;
+                : (double)shardStats.getCacheSize() / totalSize;
```

Comment:
This is indeed wrong, good catch. However, I would prefer the fix to be more explicit about the casting so that it's not misinterpreted as `(double) (shardStats.getCacheSize() / totalSize)`. I'm not saying that this is how the compiler would interpret, rather I'm thinking of the human reader. Therefore, I would prefer something more explicit like:

```diff
--- a/core/src/main/java/org/elasticsearch/indices/IndicesQueryCache.java
+++ b/core/src/main/java/org/elasticsearch/indices/IndicesQueryCache.java
@@ -103,7 +103,7 @@ public class IndicesQueryCache extends AbstractComponent implements QueryCache,
         }
         final double weight = totalSize == 0
                 ? 1d / stats.size()
-                : shardStats.getCacheSize() / totalSize;
+                : ((double) shardStats.getCacheSize()) / totalSize;
         final long additionalRamBytesUsed = Math.round(weight * sharedRamBytesUsed);
         shardStats.add(new QueryCacheStats(additionalRamBytesUsed, 0, 0, 0, 0));
         return shardStats;
```

Just an extra set of parentheses so there's no chance of confusion.